src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -281,6 +281,25 @@ internal static bool IsBinary(string text)
         public abstract int Length { get; }
 
         /// <summary>
+        /// The size of the storage representation of the text (in characters).
+        /// This can differ from length when storage buffers are reused to represent fragments/subtext.
+        /// </summary>
+        internal virtual int Size
```

Comment:
So can it still be virtual but make it StorageKey.Length? I think that will simplify ChangedText at least.